### ISSUE TITLE
Add more top-level extensions for scopes

### ISSUE THREAD

**Issue by ZacSweers** [uber/AutoDispose#347]
Right now we add scope extensions for scopable types, but let's go the whole 9 yards and add overloads for the common cases to save folks time

so instead of just offering

```kotlin
observable.autoDisposable(AndroidLifecycleScopeProvider.from(this)).subscribe()
```

Let's do this too

```kotlin
observable.autoDisposable(this).subscribe()
```

And do the same for other extension points. Only concern I'd have is possibly conflicting imports if the extensions are in different packages (but easy to test that).

**Comment by shaishavgandhi05** [comment]
Can also add an overload for the `untilEvent`:
```kotlin
observable.autoDisposable(this, ON_DESTROY).subscribe()
```

**Comment by ZacSweers** [comment]
Yep! And can make it have a default value too

**Comment by shaishavgandhi05** [comment]
Resolved in #348
